assets/js/base/hooks/use-query-state.js

```diff
@@ -52,8 +52,16 @@ export const useQueryStateByKey = ( context, queryKey ) => {
 		},
 		[ context, queryKey ]
 	);
+
 	const { setQueryValue } = useDispatch( storeKey );
-	return [ queryValue, setQueryValue ];
+	const setQueryValueByKey = useCallback(
+		( value ) => {
+			setQueryValue( context, queryKey, value );
+		},
+		[ context, queryKey ]
+	);
+
+	return [ queryValue, setQueryValueByKey ];
```

Comment:
Great fix 👍, can you extract this to it's own pull so it can get upstream quicker?  Also tests will need updated.